packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -288,6 +301,82 @@ export interface AutoScalingGroupProps extends CommonAutoScalingGroupProps {
   readonly role?: iam.IRole;
 }
 
+/**
+ * Group metrics that an Auto Scaling group sends to Amazon CloudWatch.
+ */
+export class GroupMetric {
+
+  /**
+   * All available GroupMetrics.
+   */
+  public static readonly ALL = new Array<GroupMetric>();
+
+  /**
+   * The minimum size of the Auto Scaling group
+   */
+  public static readonly MIN_SIZE = new GroupMetric('GroupMinSize');
+
+  /**
+   * The maximum size of the Auto Scaling group
+   */
+  public static readonly MAX_SIZE = new GroupMetric('GroupMaxSize');
+
+  /**
+   * The number of instances that the Auto Scaling group attempts to maintain
+   */
+  public static readonly DESIRED_CAPACITY = new GroupMetric('GroupDesiredCapacity');
+
+  /**
+   * The number of instances that are running as part of the Auto Scaling group
+   * This metric does not include instances that are pending or terminating
+   */
+  public static readonly IN_SERVICE_INSTANCES = new GroupMetric('GroupInServiceInstances');
+
+  /**
+   * The number of instances that are pending
+   * A pending instance is not yet in service, this metric does not include instances that are in service or terminating
+   */
+  public static readonly PENDING_INSTANCES = new GroupMetric('GroupPendingInstances');
+
+  /**
+   * The number of instances that are in a Standby state
+   * Instances in this state are still running but are not actively in service
+   */
+  public static readonly STANDBY_INSTANCES = new GroupMetric('GroupStandbyInstances');
+
+  /**
+   * The number of instances that are in the process of terminating
+   * This metric does not include instances that are in service or pending
+   */
+  public static readonly TERMINATING_INSTANCES = new GroupMetric('GroupTerminatingInstances');
+
+  /**
+   * The total number of instances in the Auto Scaling group
+   * This metric identifies the number of instances that are in service, pending, and terminating
+   */
+  public static readonly TOTAL_INSTANCES = new GroupMetric('GroupTotalInstances');
+
+  /**
+   * The name of the group metric
+   */
+  public readonly name: string;
+
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+
+/**
+ * A collection of group metrics
+ */
+export interface MetricsCollection {
+
+  /**
+   * The list of group metrics to monitor
+   */
+  readonly metrics: GroupMetric[];
+}
```

Comment:
why add this additional wrapper interface?

Why not just -

```ts
readonly groupMetricsCollections?: GroupMetric[];
```

in `CommonAutoScalingGroupProps`?

Reply:
Yes, I was deliberating myself on that one, here is what I was thinking, since the property is a list of lists, without the wrapper, this will be:

```ts
  readonly groupMetricsCollections?: GroupMetric[][];
```

And the usage:

```ts
  groupMetricsCollections: [
    [
      GroupMetric.DESIRED_CAPACITY
    ],
    [
      GroupMetric.MIN_SIZE,
      GroupMetric.MAX_SIZE
    ]
  ],
```
Is it less discoverable/ergonomic than:

```ts
  groupMetricsCollections: [
    {
      metrics: [ GroupMetric.DESIRED_CAPACITY ]
    },
    {
      metrics: [ GroupMetric.MIN_SIZE, GroupMetric.PENDING_INSTANCES ]
    },
  ],
```

Reply:
Can you explain why this has to be list of lists? It can be just a single list.

Reply:
Users can defined different collections, potentially with different granularity (as different granularity means different cost), It's true that now only one value is allowed which is why I don't expose it, adding it in the future will not be a breaking change. However changing from a single list to multiple lists will be a breaking change. 
The CloudFormation [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html)

"Type: List of MetricsCollection"

It makes sense for users to want to define different set of group metrics based on their type, for example "group size metrics" group that will only include the metrics  "group max size" and "group min size".

Reply:
Ah, I now see why you're trying to do that. But the ergonomics today (with only one granularity) isn't great. From the user's perspective, it's currently a list of lists and it's not clear why one will split it up or if it even matters.

Can we come up with a different way of specifying these metrics that can keep backwards compatibility and be ergonomic today?

Reply:
Added a `GroupMetrics` Class, now the API is:

```ts
new autoscaling.AutoScalingGroup(stack, 'ASG', {
  groupMetrics: [new autoscaling.GroupMetrics()],
  // ...
});
```
If in the future we need to add parameters such as `granularity` we can add `with`:
```ts
new GroupMetrics().with()
```